src/main/java/ro/utcn/sd/alexh/assignment1/controller/ConsoleController.java

```diff
@@ -0,0 +1,206 @@
+import ro.utcn.sd.alexh.assignment1.service.QuestionManagementService;
+import ro.utcn.sd.alexh.assignment1.service.TagManagementService;
+import ro.utcn.sd.alexh.assignment1.service.UserManagementService;
+
+import java.sql.Timestamp;
+import java.util.*;
+
+@Component
+@RequiredArgsConstructor
+public class ConsoleController implements CommandLineRunner {
+
+    private final Scanner scanner = new Scanner(System.in);
+    private final QuestionManagementService questionManagementService;
+    private final UserManagementService userManagementService;
+    private final TagManagementService tagManagementService;
+
+    @Override
+    public void run(String... args) throws Exception {
+        System.out.println("\nWelcome to the Poorly Implemented Stack Overflow system!\n");
+
+        handleListQuestions();
+
+        boolean done = false;
+        while (!done) {
+            System.out.println("\nEnter a command: ");
+            String command = scanner.nextLine().trim();
+            try {
+                done = handleCommand(command);
+            } catch (QuestionNotFoundException e) {
+                System.out.println("The question with the given ID was not found!");
+            } catch (AnswerNotFoundException e) {
+                System.out.println("The answer with the given ID was not found!");
+            } catch (UserNotFoundException e) {
+                System.out.println("The user with the given ID was not found!");
+            }
+        }
+    }
+
+    private boolean handleCommand(String command) {
+        System.out.println("Your command = " + command);
+        switch (command) {
+            case "list questions":
+                handleListQuestions();
+                return false;
+            case "add question":
+                handleAddQuestion();
+                return false;
+            case "register":
+                handleRegisterUser();
+                return false;
+            case "login":
+                handleLogin();
+                return false;
+            case "logout":
+                handleLogout();
+                return false;
+            case "list questions -tag":
+                handleListQuestionsByTag();
+                return false;
+            case "list questions -text":
+                handleListQuestionsByText();
+                return false;
+            case "exit":
+                return true;
+            default:
+                System.out.println("Unknown command. Try again.");
+                return false;
+        }
+    }
+
+    private void handleListQuestionsByText() {
+        List<Question> questions = questionManagementService.listQuestions();
+        List<Question> questionsCopy = questions.subList(0, questions.size());
+        Collections.reverse(questionsCopy);
+
+        String text = input("Text = ").toLowerCase();
+
+        for (Question question : questionsCopy) {
+            if (question.getTitle().toLowerCase().contains(text)) {
```

Comment:
You are filtering the questions here, in the controller; whereas you should do this filtering in the repository (where you actually already have a method for doing this).